README.md

```diff
@@ -10,4 +10,41 @@ See also [pending pull requests](https://github.com/OCFL/extensions/pulls) for e
+provides both more effective documention of an OCFL structure but allows the implementation of generic extension code that
+covers a wider variety of use cases. Parameters MUST have be single valued. For each parameter the following properties should
+be defined:    
+
+* Name: A short name for the parameter. Since this has the potential to be used as part of programmatic access the name MUST
+comply with the Javascript restrictions on identifiers (The first character must be a letter, an underscore, or a dollar sign. 
+Subsequent characters may be letters, digits, underscores, or dollar signs.) and SHOULD be shorter than 127 characters. The
+length limit is based on a survey of the defaults for various JSON parsers. 
+* Description: A brief description of the function of the parameter. This should be expanded in the main description of the
+extension which MUST reference all the parameters.
+* Type: Data type for the parameter. In order to allow validation and limit the scope for implementation specific variations,
+parameters are typed.
+  * integer - may be signed or not as specified in the range parameter.
+  * string - aligned with JSON strings, these should be UTF-8 encoded and avoid control characters.  
+  * enumerated - one of an ordered set of labels which MUST conform to the same limitations as parameter names.  
+  * boolean - may have the values false or true (lower case and unquoted as in JSON)
+  * For integer parameters the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
+  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, try to keep strings shorter than 4095 characters.
+  * For enumerated parameters, the range is a comma separated ordered list of labels that are valid for the parameter. Enumerated parameters are case sensitive and MUST always be quoted, so they are JSON strings. 
+* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory 
+
+## Referencing Parameters
+
+Wherever a parameterised extension is referenced, include any parameters in an accompanying JSON file. If using an extensions directory, the JSON file MUST be named for the extension and included in the directory. For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
+
+    "0000-example-extension.md": {  
+        "firstExampleParameter": 12,  
+        "secondExampleParameter": "Hello",  
+        "thirdExampleParameter": "Green"  
+    }
+If, instead, the extension is referenced in a JSON file (e.g. additional digests in a manifest) then these can be included immediately following the reference. However, there MUST only be one parameter set defined for each extension.
```

Comment:
I know you haven't resolved [issue 6](https://github.com/OCFL/extensions/issues/6) yet, but perhaps you can give me preview of what you're thinking in regards to the "inline" parameter definitions in the case of additional digests?

My assumption, based on reading the text is that it'd look something like this:

```
{
    "id": "obj1",
    "type": "https://ocfl.io/1.0/spec/#inventory",
    "digestAlgorithm": "sha512_256",
    "0001-digest-algorithms.md": {
        "sha512_256": true
    },
    "head": "v1",
    "contentDirectory": "content",
    "fixity": {},
    "manifest": {
        "5a..e1": [
            "v1/content/test.txt"
        ]
    },
    "versions": {
        "v1": {
            "created": "2020-05-20T16:26:11Z",
            "message": "initial commit",
            "state": {
                "5a..e1": [
                    "test.txt"
                ]
            }
        }
    }
}
```

My understanding is that this would not be allowed by the spec because additional properties cannot be inserted into the inventory. So, I'm guessing I'm missing something in regards to how you're envisioning the parameterization working in this case and am curious what it would look like.

Reply:
Not necessarily. The inventory definition states what MUST and MAY be there but does not preclude other entries. This is very much a first cut on parameterization, however. I think we need to work through some more cases before fixing things in stone. But I was thinking along the lines of your example...

Reply:
I could have sworn that I'd read in the spec that it forbid additional properties in inventory files, but, you're right, I couldn't find that language when I looked for it now.

Reply:
I would be inclined to remove the last two sentences ("If, instead, the extension is referenced in a JSON file..."), as I am not sure we have a clear use case for inline extension parameterization at this point. The example use case here (digest algorithms) is pretty well covered in the spec: https://ocfl.io/draft/spec/#digests - paragraph no.3

Reply:
@awoods I was about to suggest the same (drops last two sentences) since it opens a can of worms. Although, I also note that the presence of para. 3 in digests also kind of means that the extensions mechanism needs to be defined for 1.0 release.

Reply:
...actually, it makes sense keep the bit about only one parameter set.